integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonGlobalDictionaryRDD.scala

```diff
@@ -83,7 +84,12 @@ case class PrimitiveParser(dimension: CarbonDimension,
 
   def parseString(input: String): Unit = {
     if (hasDictEncoding && input != null) {
-      set.add(input)
+      if (set.size < CarbonCommonConstants.MAX_EXTERNAL_DICTIONARY_SIZE) {
```

Comment:
This check will be done for each value for each column...please ensure there is no performance degradation in dictionary generation due to this check...try with some 100 columns with and without this change